File: LuckyNumberApp/app/src/main/java/com/example/luckynumberapp/MainActivity.java
package com.example.luckynumberapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText editText=findViewById(R.id.editText);
        Button button=findViewById(R.id.button);

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String username=editText.getText().toString();

                //Explicit Intent
                Intent intent=new Intent(getApplicationContext(), MainActivity2.class);
                intent.putExtra("name",username);
                startActivity(intent);
            }
        });
    }

//    public void goToSecond(){
//        Intent intent=new Intent(this, MainActivity2.class);
//        startActivity(intent);
//    }
}

File: LuckyNumberApp/app/src/main/java/com/example/luckynumberapp/MainActivity.kt
package com.example.luckynumberapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val et:EditText=findViewById(R.id.editText);
        val btn:Button=findViewById(R.id.button);

        btn.setOnClickListener(){
            val edittext=et.text;
            val i:Intent=Intent(this,MainActivity2::class.java);
            i.putExtra("name",edittext);
            startActivity(i);
        }
    }
}